### ISSUE TITLE
Events: Trigger may need to be a struct

### ISSUE THREAD

**Issue by paulflang** [LCSB-BioCore/SBML.jl#202]
https://github.com/LCSB-BioCore/SBML.jl/blob/3254ddeeee71689776c1eca7fff3ce23719d5a01/src/structs.jl#L297

I just looked at the specs of events (p81) and I think we need to make trigger a separate struct. Sth like

```julia
struct EventTrigger
    initial_value::Bool
    persistent::Bool
    math::Maybe{SBML.Math} = nothing
end
```

Note to self: Due to how event handling currently works in DifferentialEquations.jl, I think SBMLTk would have to throw an error if `persistent=false`. But perhaps we could translate `initial_value=false` to `ContinuousCalleback(...; initialise=(a,b,c,d) -> -1.)` and `initial_value=true` to `initialise=(a,b,c,d) ->1.`.

**Comment by exaexa** [comment]
Sure, that sounds plausible. Corresponds to `Trigger_t` from here, right?
https://synonym.caltech.edu/software/libsbml/5.18.0/docs/formatted/c-api/class_trigger__t.html

(Also, if I get the implicitst hints in the documentation right, the trigger attaches to event, and each event may have 0 or 1 triggers, right? Or is it always 1 trigger?)

**Comment by exaexa** [comment]
( Response to note to self: converting level-triggers (math here) to edge-triggers (events) is wrong™ but it looks like there aren't any other possibilities. :D )

**Comment by paulflang** [comment]
> Sure, that sounds plausible. Corresponds to Trigger_t from here, right?
https://synonym.caltech.edu/software/libsbml/5.18.0/docs/formatted/c-api/class_trigger__t.html

> (Also, if I get the implicitst hints in the documentation right, the trigger attaches to event, and each event may have 0 or 1 triggers, right? Or is it always 1 trigger?)

Yes, both is correct.

**Comment by paulflang** [comment]
> ( Response to note to self: converting level-triggers (math here) to edge-triggers (events) is wrong™ but it looks like there aren't any other possibilities. :D )

What is a level-trigger and what is an edge-trigger?

**Comment by exaexa** [comment]
> What is a level-trigger and what is an edge-trigger?

https://electricalbaba.com/edge-triggering-and-level-triggering/

in short, edge triggering is complicated in electrical engineering but if you want anything to be precise, you _need_ edge triggers in computers.

> Yes, both is correct.
b b b but the statements contradict themselves :sob:

**Comment by exaexa** [comment]
....OK it's 0-1:
https://github.com/sbmlteam/libsbml/blob/development/src/sbml/Event.h#L1487

**Comment by paulflang** [comment]
> b b b but the statements contradict themselves 😭

Ah sorry, I meant "Corresponds to Trigger_t from here" is correct, and "each event may have 0 or 1 triggers" is also correct.

**Comment by giordano** [comment]
Incidentally fixed by #11: [`33a385d` (#143)](https://github.com/LCSB-BioCore/SBML.jl/pull/143/commits/33a385dcb83d74a02f6ad1e5b0a2059abfd9268e)
